File: yunva17/프로그래머스/Lv1/하샤드_수.js
function solution(x) {
  var answer = true;
  var nString = x.toString();
  var nSum = 0;

  for (var i = 0; i < nString.length; i++) {
    nSum += parseInt(nString[i]);
  }

  if (x % nSum === 0) {
    answer = true;
  } else {
    answer = false;
  }

  return answer;
}

File: overthename/Programmers/level1/하샤드수.java
import java.util.*;
class Solution {
	public boolean solution(int x) {
		boolean answer = true;

		String number = Integer.toString(x);

		int sum=0;
		for(int i=0; i<number.length();i++){
			sum+= Integer.parseInt(String.valueOf(number.charAt(i)));
		}

		if(x%sum==0){
			return true;
		}else{
			return false;
		}
	}
}